src/brackets.js

```diff
@@ -114,6 +114,23 @@ define(function (require, exports, module) {
             return CodeMirror;
         }
     });
+    
+    // DEPRECATED: Mustache.compile has been deprecated since Mustache 0.8,
+    // but for now we use a polyfill for it so as to avoid breaking extensions in the
+    // interim.
+    Object.defineProperty(window.Mustache, "compile", {
```

Comment:
This is great. I am not sure how prominent `compile` is, but it is good to warn and tag it a deprecated.  @ficristo is right about letting extension authors this change.  @MarcelGerber might be able to give us some guidance on this. :)